balancer-js/src/modules/joins/graph.ts

```diff
@@ -151,9 +175,13 @@ export class PoolGraph {
       if (linearPool.mainIndex === undefined)
         throw new Error('Issue With Linear Pool');
 
+      const mainTokenDecimals =
+        linearPool.tokens[linearPool.mainIndex].decimals ?? 18;
```

Comment:
I'm not sure how likely is for this not to be defined, but maybe throwing an error is safer than falling back to 18 decimals.